src/main/java/com/webapp/TextBook/sharedFiles/ValidationBindingHelper.java

```diff
@@ -0,0 +1,181 @@
+
+    private static final String failedValidationStatusMessage = "Input missing or invalid";
+    private static final String INTERNAL_VALIDATION_ERROR = "Internal Error: please contact IT.";
+
+    // creates local final strings for error and general errors key
+    private static final String ERRORS_KEY = "Errors";
+    private static final String GENERAL_ERRORS_KEY = "GeneralError";
+
+    public static Pair<Boolean,JSONObject> validationBindingHandler(@NotNull final BindingResult result, @NotNull final ModelMap map) throws JSONException {
+        JSONObject data = new JSONObject();
+        Boolean returnResult = null;
+        try {
+
+
+            if (result.hasErrors()) {
+                //binding errors present, load in Status/Error messages and return to login
+                if (result.getErrorCount() != 1) {
+                    throw new IllegalArgumentException("Internal Error: (WebControllerValidationBindingHelper" +
+                        ":validationBindingHandler) - " +
+                        "View model Retaining Multiple validation Bindings. " +
+                        "View model should only retain one class/type validation");
+                }
+                ObjectError loginUserInfoError = result.getAllErrors().get(0);
+
+
+                data.put("StatusMessage", failedValidationStatusMessage);
+
+                // invoke helper to set error state
+                ValidationBindingHelper.handleErrorMessageType(loginUserInfoError.getDefaultMessage(), data);
+
+                map.addAttribute("Data", data);
+                returnResult = false;
+            }
+            else{
+                returnResult = true;
+            }
+        }
+        catch(IllegalArgumentException i){
+            System.out.println();
+            returnResult = false;
+        }
+        catch(Exception e){
+            System.out.println("Internal Error: (WebControllerValidationBindingHelper" +
+                    ":validationBindingHandler) - " +
+                    "");
+            returnResult = false;
+        }
+        finally {
+            return new Pair<Boolean, JSONObject>(returnResult,data);
+        }
+    }
+
+    // assist in validation execution for api validation
+    // note: json exception guarantee not to occur
+    public static boolean handleApiValidationBindingForJsonOutput(@NotNull final Optional<String> apiValidationResult, @NotNull final JSONObject jsonObject) throws JSONException{
+
+        // if optional is empty then return success result, else invoke helper to set error state
+        if(apiValidationResult.isEmpty()) {
+           ValidationBindingHelper.handleSuccessValidationErrorMessage(jsonObject);
+           return true;
+        }
+
+        ValidationBindingHelper.handleErrorMessageType(apiValidationResult.get(), jsonObject);
+        return false;
+    }
+
+    // todo: create handler for conjoined optional validation results on stream-filter predicates
```

Comment:
todo can be disregarded but some serious doc is needed here (all methods need it but there should be a comment above requesting it for the whole class)